### PR TITLE
Update test cases for phase service

### PR DESCRIPTION
**Pull request by NereusWB922**
### Summary:

Update test cases for phase service to match changes made in #255 

#### Type of change:

- 🧪 Tests Update

### Changes Made:

- Refactor test cases for `setRepository`
- Add test cases for `setupFromUrl` method

### Proposed Commit Message:

```
Refactor and add test cases for phase service

Phase service has been updated. 

Let's refactor and add test cases to match the changes.
```

### CODE REVIEW

File: tests/services/phase.service.spec.ts
Comment (luminousleek) [inline tests/services/phase.service.spec.ts:197]: Wondering if we should test if this throws the `ErrorMessageService.repositoryNotPresentMessage` error instead of a generic error.

Comment (NereusWB922) [reply tests/services/phase.service.spec.ts:189]: Yes, I think it's a good idea to test for the specific error.

**Review by Arif-Khalid — COMMENTED** [review]
Tests are comprehensive and are effectively refactored for usage in WATcher instead of CATcher.
Tests run successfully.
Should just wait for senior's approval before merging.

**Review by luminousleek — APPROVED** [review]
LGTM, thanks for working on refactoring all the tests!

**Pull request merged**
